sky/provision/azure/instance.py

```diff
@@ -156,17 +600,32 @@ def terminate_instances(
     assert provider_config is not None, cluster_name_on_cloud
 
     resource_group_client = azure.get_client('resource', subscription_id)
-    delete_resource_group = get_azure_sdk_function(
+    delete_resource_group = _get_azure_sdk_function(
         client=resource_group_client.resource_groups, function_name='delete')
 
-    delete_resource_group(resource_group, force_deletion_types=None)
+    try:
+        delete_resource_group(resource_group, force_deletion_types=None)
+    except azure.exceptions().ResourceNotFoundError as e:
+        if 'ResourceGroupNotFound' in str(e):
+            logger.warning(f'Resource group {resource_group} not found. Skip '
+                           'terminating it.')
+            return
+        raise
 
 
-def _get_vm_status(compute_client: 'azure_compute.ComputeManagementClient',
-                   vm_name: str, resource_group: str) -> str:
-    instance = compute_client.virtual_machines.instance_view(
-        resource_group_name=resource_group, vm_name=vm_name).as_dict()
-    for status in instance['statuses']:
+def _get_instance_status(
+        compute_client: 'azure_compute.ComputeManagementClient', vm,
+        resource_group: str) -> Optional[AzureInstanceStatus]:
+    try:
+        instance = compute_client.virtual_machines.instance_view(
+            resource_group_name=resource_group, vm_name=vm.name)
```

Comment:
should we use the `_get_azure_sdk_function` for the `instance_view` function as well?